File: frontend/src/types/Lab.ts
// src/types/Lab.ts
export interface Lab {
    id: number;
    name: string;
    location: string;
    capacity: number;
    description?: string;
}

export interface LabCreateRequest {
    name: string;
    location: string;
    capacity: number;
    description?: string;
}

File: backend/src/main/java/com/example/labsystem/domain/lab/Lab.java
package com.example.labsystem.domain.lab;

import jakarta.persistence.*;
import lombok.*;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Entity
@Table(name = "labs")
public class Lab {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String name;
    private String location;
    private Integer capacity;

    @Column(length = 255)
    private String description;
}
